src/index.test.tsx

```diff
@@ -0,0 +1,238 @@
+
+const IndexCell = () => {
+  const index = useIndexContext();
+  return <Fragment>{index + 1}.</Fragment>;
+};
+
+const data = [
+  {
+    id: '1',
+    name: 'Gloves',
+    price: 544,
+    count: 5,
+    color: 'plum',
+  },
+  {
+    id: '2',
+    name: 'Salad',
+    price: 601,
+    count: 6,
+    color: 'turquoise',
+  },
+  {
+    id: '3',
+    name: 'Keyboard',
+    price: 116,
+    count: 1,
+    color: 'silver',
+  },
+];
+
+describe('ctablex', () => {
+  it('should render table', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column>
+            <IndexCell />
+          </Column>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+          <Column header="Price" accessor="price">
+            <StringCell />
+          </Column>
+          <Column header="Count" accessor="price">
+            <StringCell />
+          </Column>
+          <Column header="Total" accessor={(r: any) => r.price * r.count}>
+            <StringCell />
+          </Column>
+          <Column header="Color" accessor="color">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should provide index context', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column>
+            <IndexCell />
+          </Column>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should use custom adapter', () => {
+    const adapter = {
+      table: (props: PropsWithChildren<{}>) => {
+        return (
+          <table style={{ backgroundColor: '#ccc' }}>{props.children}</table>
+        );
+      },
+    };
+
+    const c = (
+      <TableUIAdapterContextProvider value={adapter}>
+        <DataTable data={data}>
+          <Columns>
+            <Column header="Name" accessor="name">
+              <StringCell />
+            </Column>
+          </Columns>
+          <Table>
+            <TableHeader>
+              <HeaderRow />
+            </TableHeader>
+            <TableBody>
+              <Rows>
+                <Row />
+              </Rows>
+            </TableBody>
+          </Table>
+        </DataTable>
+      </TableUIAdapterContextProvider>
+    );
+    const { unmount } = render(c);
```

Comment:
```suggestion
expect(table).toHaveStyle({backgroundColor: '#ccc'})
```

- Also you can assign a `role` or `data-testid` to table in order to  find it easier.
- And run this test for all the adapter components